cnf-certification-test/lifecycle/suite.go

```diff
@@ -584,23 +581,36 @@ func TestPodTolerationBypass(env *provider.TestEnvironment) {
 	testhelper.AddTestResultLog("Non-compliant", podsWithRestrictedTolerationsNotDefault, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 func testStorageRequiredPods(env *provider.TestEnvironment) {
-	oc := clientsholder.GetClientsHolder()
-	var nonCompliantPods []*provider.Pod
-	for _, pod := range env.Pods {
-		for _, podOwner := range pod.ObjectMeta.GetOwnerReferences() {
-			if podOwner.Kind != statefulSet {
-				continue
-			}
-			statefulset, err := oc.K8sClient.AppsV1().StatefulSets(pod.Namespace).Get(context.TODO(), podOwner.Name, apiv1.GetOptions{})
-			if err != nil {
-				tnf.ClaimFilePrintf(err.Error())
+	var podsWithLocalStorage []string
+	for _, put := range env.Pods {
+		for pvIndex := range put.Spec.Volumes {
+			// Skip any nil persistentClaims.
+			if put.Spec.Volumes[pvIndex].PersistentVolumeClaim == nil {
 				continue
 			}
-			if statefulset.Spec.ServiceName == localStorage {
-				nonCompliantPods = append(nonCompliantPods, pod)
-				continue
+			// We have the list of pods/volumes/claims.
+			// Look through the storageClass list for a match.
+
+			var StorageClasses = env.StorageClasses
+			var Pvc = env.PersistentVolumeClaims
+			for i := range Pvc {
+				if Pvc[i].Name == put.Spec.Volumes[pvIndex].PersistentVolumeClaim.ClaimName {
+					for j := range StorageClasses {
+						if StorageClasses[j].Provisioner == "kubernetes.io/no-provisioner" {
+							tnf.ClaimFilePrintf("%s has been found to use a local storage enabled storageClass.\n Pvc_name: %s, Storageclass_name: %s, Provisionner_name: %s", put.String(), put.Spec.Volumes[pvIndex].PersistentVolumeClaim.ClaimName,
+								StorageClasses[j].Name, StorageClasses[j].Provisioner)
+							podsWithLocalStorage = append(podsWithLocalStorage, put.String())
+							// Break out of the storageList loop.
+							continue
+						}
+						tnf.ClaimFilePrintf("%s has been not found to use a local storage enabled storageClass.\n Pvc_name: %s, Storageclass_name: %s, Provisionner_name: %s", put.String(), put.Spec.Volumes[pvIndex].PersistentVolumeClaim.ClaimName,
```

Comment:
I don't believe we need to print when we found a valid pod, only invalid should be in the claim file print.

Reply:
Yes but look the comment of @edcdavid 
I understood that need print information also in this case.